tasks/managed/rh-sign-image-cosign/rh-sign-image-cosign.yaml

```
@@ -251,45 +251,66 @@ spec:
             COMPONENT_NAME=$(jq -r ".components[${COMPONENTS_INDEX}].name" "${SNAPSHOT_PATH}")
             echo "Processing component ${COMPONENT_NAME}"
 
-            # Get public image references
-            INTERNAL_CONTAINER_REF=$(jq -r ".components[${COMPONENTS_INDEX}].repository" "${SNAPSHOT_PATH}")
-            rh_registry_repo=$(jq -r ".components[${COMPONENTS_INDEX}][\"rh-registry-repo\"]" "${SNAPSHOT_PATH}")
-            registry_access_repo=$(jq -r ".components[${COMPONENTS_INDEX}][\"registry-access-repo\"]" \
-              "${SNAPSHOT_PATH}")
-            repository="${rh_registry_repo#*/}"
-
-            # Sign rh-registry-repo references (always) and registry-access-repo references
-            # (only if signatures for this registry are required)
-            REGISTRY_REFERENCES=("${rh_registry_repo}")
-            if grep -q "^${repository}$" "${SIGN_REGISTRY_ACCESS_FILE}"; then
-              REGISTRY_REFERENCES+=("${registry_access_repo}")
-            fi
+            COMPONENT=$(jq -c ".components[${COMPONENTS_INDEX}]" "${SNAPSHOT_PATH}")
 
             # Check if image is manifest list
             BUILD_CONTAINER_IMAGE=$(jq -r ".components[${COMPONENTS_INDEX}].containerImage" "${SNAPSHOT_PATH}")
             DIGEST="${BUILD_CONTAINER_IMAGE/*@}"
             IMAGE=$(skopeo inspect --raw "docker://${BUILD_CONTAINER_IMAGE}")
             MEDIA_TYPE=$(echo "$IMAGE" | jq -r '.mediaType')
-            TAGS=$(jq -r ".components[${COMPONENTS_INDEX}].tags|.[]" "${SNAPSHOT_PATH}")
             LIST=0
             if [ "$MEDIA_TYPE" = "application/vnd.docker.distribution.manifest.list.v2+json" ]; then LIST=1; fi
             if [ "$MEDIA_TYPE" = "application/vnd.oci.image.index.v1+json" ]; then LIST=1; fi
 
-            # Collect data for signing
-            # Sign each manifest in the manifest list
-            if [ $LIST -eq 1 ]; then
-                for REGISTRY_REF in "${REGISTRY_REFERENCES[@]}"; do
-                    for MDIGEST in $(echo "$IMAGE" | jq -r '.manifests[]|.digest'); do
-                        for TAG in $TAGS; do
-                            to_sign+=("${REGISTRY_REF}:${TAG}@${MDIGEST}#${INTERNAL_CONTAINER_REF}")
+            # Process repositories array (apply-mapping guarantees this exists)
+            NUM_REPOSITORIES=$(jq '.repositories | length' <<< "$COMPONENT")
+            for (( REPO_INDEX=0; REPO_INDEX<NUM_REPOSITORIES; REPO_INDEX++ )); do
```

Just a style nitpick, we usually use i, j and so on for the loop indexes.
```suggestion
            for (( i = 0; i < NUM_REPOSITORIES; i++ )); do
```

---

Valid point, thanks for the suggestion, fixed :)